jmbitcoin/jmbitcoin/secp256k1_main.py

```diff
@@ -575,56 +225,52 @@ def ecdsa_raw_verify(msg, pub, sig, usehex, rawmsg=False):
-    except:
+    except Exception as e:
         return False
     return retval
 
-def estimate_tx_size(ins, outs, txtype='p2pkh'):
-    '''Estimate transaction size.
-    The txtype field as detailed below is used to distinguish
-    the type, but there is at least one source of meaningful roughness:
-    we assume the output types are the same as the input (to be fair,
-    outputs only contribute a little to the overall total). This combined
-    with a few bytes variation in signature sizes means we will expect,
-    say, 10% inaccuracy here.
-
-    Assuming p2pkh:
-    out: 8+1+3+2+20=34, in: 1+32+4+1+1+~73+1+1+33=147,
-    ver:4,seq:4, +2 (len in,out)
-    total ~= 34*len_out + 147*len_in + 10 (sig sizes vary slightly)
-    Assuming p2sh M of N multisig:
-    "ins" must contain M, N so ins= (numins, M, N) (crude assuming all same)
-    74*M + 34*N + 45 per input, so total ins ~ len_ins * (45+74M+34N)
-    so total ~ 34*len_out + (45+74M+34N)*len_in + 10
-    Assuming p2sh-p2wpkh:
-    witness are roughly 3+~73+33 for each input
-    (txid, vin, 4+20 for witness program encoded as scriptsig, 4 for sequence)
-    non-witness input fields are roughly 32+4+4+20+4=64, so total becomes
-    n_in * 64 + 4(ver) + 4(locktime) + n_out*34
-    Assuming p2wpkh native:
-    witness as previous case
-    non-witness loses the 24 witnessprogram, replaced with 1 zero,
-    in the scriptSig, so becomes:
-    n_in * 41 + 4(ver) + 4(locktime) +2 (len in, out) + n_out*34
-    '''
-    if txtype == 'p2pkh':
-        return 10 + ins * 147 + 34 * outs
-    elif txtype == 'p2sh-p2wpkh':
-        #return the estimate for the witness and non-witness
-        #portions of the transaction, assuming that all the inputs
-        #are of segwit type p2sh-p2wpkh
-        # Note as of Jan19: this misses 2 bytes (trivial) for len in, out
-        # and also overestimates output size by 2 bytes.
-        witness_estimate = ins*109
-        non_witness_estimate = 4 + 4 + outs*34 + ins*64
-        return (witness_estimate, non_witness_estimate)
-    elif txtype == 'p2wpkh':
-        witness_estimate = ins*109
-        non_witness_estimate = 4 + 4 + 2 + outs*31 + ins*41
-        return (witness_estimate, non_witness_estimate)
-    elif txtype == 'p2shMofN':
-        ins, M, N = ins
-        return 10 + (45 + 74*M + 34*N) * ins + 34 * outs
-    else:
-        raise NotImplementedError("Transaction size estimation not" +
-                                  "yet implemented for type: " + txtype)
+class JMCKey(bytes, CKeyBase):
+    """An encapsulated private key.
+    This subclasses specifically for JM's own signing code.
+
+    Attributes:
+
+    pub           - The corresponding CPubKey for this private key
+    secret_bytes  - Secret data, 32 bytes (needed because subclasses may have trailing data)
+
+    is_compressed() - True if compressed
+
+    """
+
+    def __init__(self, b):
+        CKeyBase.__init__(self, b, compressed=True)
+
+    def is_compressed(self):
```

Comment:
If `CKeyBase` works correctly, is_compressed is passed as `SECP256K1_EC_COMPRESSED` to `secp256k1_ec_pubkey_serialize`, and the pubkey is saved in `cls.__pub`. Unless someone messes with `__pub`, it will be compressed throughout the lifetime. If it becomes not compressed for some reason, I don't see how masking that fact by returning true here would help

Reply:
Thanks, this is as per the next comment, addressed in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/1e2fc3848295a05f4ee685fd435f277c378acadc